server/src/main/java/org/elasticsearch/transport/TransportRequest.java

```diff
@@ -16,16 +16,6 @@
 import java.io.IOException;
 
 public abstract class TransportRequest extends TransportMessage implements TaskAwareRequest {
-    public static class Empty extends TransportRequest {
-        public static final Empty INSTANCE = new Empty();
-
-        public Empty() {}
-
-        public Empty(StreamInput in) throws IOException {
-            super(in);
-        }
-    }
```

Comment:
I wonder how we can prevent this to be re-added as a refactor in future. Maybe leave a comment here?

Reply:
I'm not sure we can tbh, although it's not the end of the world if it comes back, and hopefully it causes the contributor/reviewer to pause for thought that such a basic thing doesn't already exist.